packages/@jsii/go-runtime/jsii-runtime-go/typeregistry/discover-implemented.go

```diff
@@ -0,0 +1,43 @@
+package typeregistry
+
+import (
+	"reflect"
+
+	"github.com/aws/jsii-runtime-go/api"
+)
+
+type DiscoverImplementationer interface {
+	// DiscoverImplementation determines the list of registered interfaces that
+	// are implemented by the provided type, and returns the list of their FQNs
+	// and overrides for all their combined methods and properties.
+	DiscoverImplementation(t reflect.Type) ([]api.FQN, []api.Override)
+}
+
+func (t *TypeRegistry) DiscoverImplementation(vt reflect.Type) (interfaces []api.FQN, overrides []api.Override) {
+	registeredOverrides := make(map[string]bool)
+
+	for fqn, members := range t.typeMembers {
+		iface := t.fqnToType[fqn]
+		if iface.Kind == classType || !vt.AssignableTo(iface.Type) {
+			continue
+		}
```

Comment:
This might be expensive (I don't know). I decided to KISS for now, but we should be able to add a cache layer around this to mitigate the cost.